woocommerce/woocommerce-ios
# REST API: Web view for application password authorization

Part of #9186 
Please also review https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/758.


## Description

This PR adds a new web view for authorizing application password during login to help with cases where site credential login fails when handled programmatically. Changes include:
- Networking: updated the `WordPressSite` model to decode the application password authorization URL if it exists.
- Yosemite: updated `WordPressSiteStore` and action to fetch application password authorization URL.
- UI:
  - Added a new view controller for application password authorization. This controller handles fetching the authorization URL and showing it in a web view.
  - Updated the `AuthenticationManager` to enable a new config for manual error handling of site credential login.
  - Implemented the new delegate method to show a custom alert with the entry point to application password authorization in web view.


## Testing instructions

- Attempt to log in to a self-hosted site with incorrect credentials.
- Notice that the error alert has a button to log in through WP-Admin.
- Log in with the correct credentials on the presented web view.
- Reject the authorization, the web view should be dismissed after clicking on Cancel button of the presented error alert.

The success case will be handled in a subsequent PR.


## Screenshots


https://user-images.githubusercontent.com/5533851/226259290-434eac62-3565-4124-8e7a-c6591bec5b0a.mp4



---
